integration/feature/init/src/MillInitMavenTests.scala

```diff
@@ -0,0 +1,221 @@
+    val unzipDir = os.unzip(zipFile, os.temp.dir())
+    val sourceDir = os.list(unzipDir).head
+
+    integrationTest { tester =>
+      for (p <- os.list(sourceDir)) os.move.into(p, tester.workspacePath)
+      f(tester)
+    }
+  }
+
+  def tests: Tests = Tests {
+
+    // - Junit5
+    // - maven-compiler-plugin release option
+    test("jansi") {
+      prep("https://github.com/fusesource/jansi/archive/refs/tags/jansi-2.4.1.zip") { tester =>
+        import tester._
+
+        val initRes = eval(("init", "--deps-object", "Deps"))
+        initRes.out.contains(
+          "generated 1 Mill build file(s)"
+        ) ==> true
+        initRes.isSuccess ==> true
+
+        val compileRes = eval("compile")
+        compileRes.err.contains(
+          "compiling 20 Java sources"
+        ) ==> true
+        compileRes.isSuccess ==> true
+
+        val testRes = eval("test")
+        testRes.out.contains(
+          "Test run finished: 0 failed, 1 ignored, 90 total"
+        ) ==> true
+        testRes.isSuccess ==> true
+
+        val publishLocalRes = eval("publishLocal")
+        publishLocalRes.err.contains(
+          "Publishing Artifact(org.fusesource.jansi,jansi,2.4.1)"
+        ) ==> true
+        publishLocalRes.isSuccess ==> true
+      }
+    }
+
+    // - multi-module
+    // - TestNg
+    // - maven-compiler-plugin release option
+    test("dotenv") {
+      prep("https://github.com/shyiko/dotenv/archive/refs/tags/0.1.1.zip") { tester =>
+        import tester._
+
+        val initRes = eval("init")
+        initRes.out.contains(
+          "generated 3 Mill build file(s)"
+        ) ==> true
+        initRes.isSuccess ==> true
+
+        val resolveRes = eval(("resolve", "_"))
+        Seq(
+          "dotenv",
+          "dotenv-guice"
+        ).forall(resolveRes.out.contains) ==> true
+
+        // JavaModule.JavaTests is not picking compileIvyDeps from outer module
+        val compileRes = eval("__.compile")
+        compileRes.err.contains("package com.google.inject does not exist") ==> true
+        compileRes.isSuccess ==> false
+
+        // even if compile error is fixed, TestNg version is not supported
+        // val testRes = eval("__.test")
+        // testRes.isSuccess ==> false
+      }
+    }
+
+    // - multi-module
+    // - Junit5
+    // - submodule e2e directory and artifact name differ
+    test("fastexcel") {
+      prep("https://github.com/dhatim/fastexcel/archive/refs/tags/0.18.4.zip") { tester =>
+        import tester._
+
+        val initRes = eval("init")
+        initRes.out.contains(
+          "generated 4 Mill build file(s)"
+        ) ==> true
+        initRes.isSuccess ==> true
+
+        val resolveRes = eval(("resolve", "_"))
+        Seq(
+          "fastexcel-reader",
+          "fastexcel-writer",
+          "e2e"
+        ).forall(resolveRes.out.contains) ==> true
+
+        // not sure why this happens but pom.xml has a hack to handle JPMS
+        val compileRes = eval("__.compile")
```

Comment:
The [hack](https://github.com/dhatim/fastexcel/blob/de56e786a1fe29351e2f8dc1d81b7cdd9196de4a/pom.xml#L252).

Reply:
Could you elaborate a bit on how this affects us? The comment says `Run Java8 build with JPMS on JDK9+`, but AFAIK we already run Mill CI on JDK >=11, so I thought we would be on JDK9+ already.

Reply:
This is from the [executions](https://github.com/dhatim/fastexcel/blob/de56e786a1fe29351e2f8dc1d81b7cdd9196de4a/pom.xml#L268) section.
```xml
<executions>
    <!-- First: compile all as Java9 -->
    <execution>
        <id>default-compile</id>
        <configuration>
            <release>9</release>
            <excludes>
                <!-- package-info.java is not recompiled as Java8 in the second step for an unknown reason -->
                <!-- this workaround just forces the second compilation by avoiding the first one -->
                <exclude>**/package-info.java</exclude>
            </excludes>
        </configuration>
    </execution>
    <!-- Second: recompile all but module-info as Java8 -->
    <execution>
        <id>base-compile</id>
        <goals>
            <goal>compile</goal>
        </goals>
        <configuration>
            <source>8</source>
            <target>8</target>
            <release>8</release>
            <excludes>
                <exclude>module-info.java</exclude>
            </excludes>
        </configuration>
    </execution>
</executions>
```

Reply:
Added test for [avaje-config](https://github.com/avaje/avaje-config) that also has JPMS errors.